garminworkouts/config/excelparser.py

```diff
@@ -0,0 +1,67 @@
+import pandas as pd
+import numpy as np
+
+def excel_to_yaml(filename,ftp):
+
+    df=pd.read_excel(filename)
+    df=df[df.columns.tolist()[:3]]
+    df.columns=["start","end","duration"]
+    df.reset_index(inplace=True, drop=True)
+
+    def Normalise(x,ftp=ftp):
+        if str(x) == str(np.nan):
+            return x
+        elif "%" in str(x):
+            x=str(x).replace("%","").replace(" ","").replace(",",".")
+            x=float(x)
+            return x
+        else:
+            x=float(x)/float(ftp)*100
+            x=float(x)
+            return x
+
+    df["start"]=df["start"].apply(lambda x: Normalise(x) )
+    df["end"]=df["end"].apply(lambda x: Normalise(x) )
+    df["duration"]=df["duration"].astype(str)
+
+    workout_name=filename.replace("_"," ").split(".xls")[0].split("/")[-1]
+    workout_name='name: "{}"\n'.format(workout_name)
+
+    def create_steps(i,df=df,seconds_block=10):
+        start=df.loc[i,"start"]
+        end=df.loc[i,"end"]
+        duration=df.loc[i,"duration"]
+        if str(end) == str(np.nan):
+            return '  - {{ power: {power}, duration: "{duration}" }}\n'.format(power=round(start,2), duration=duration)
+        else:
+            if start == end:
+                return '  - {{ power: {power}, duration: "{duration}" }}\n'.format(power=round(start,2), duration=duration)
+            elif start != end:
+                seconds=int(duration.split(":")[-1])+int(duration.split(":")[-2])*60
+                n_blocks=seconds/seconds_block
```

Comment:
Are you aware of maximum number of steps in the single Garmin workout? If I remember the limit is around 50.

Reply:
read about that today; will take your word for it. 
solved in 5999552